staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -527,6 +527,15 @@ type PersistentVolumeClaimSpec struct {
 	// (Alpha) Using the namespace field of dataSourceRef requires the CrossNamespaceVolumeDataSource feature gate to be enabled.
 	// +optional
 	DataSourceRef *TypedObjectReference `json:"dataSourceRef,omitempty" protobuf:"bytes,8,opt,name=dataSourceRef"`
+	// volumeAttributesClassName is the name of the VolumeAttributesClass required by the claim.
+	// If specified, the provisioner will create or update the volume with the attributes defined
+	// in the corresponding VolumeAttributesClass. Different with storageClassName, it can be
+	// changed after the claim is created but an empty string value is disallowed.
+	// More info: https://kubernetes.io/docs/concepts/storage/persistent-volumes#volumeattributesclass
+	// (Alpha) Using this field requires the VolumeAttributesClass feature gate to be enabled.
+	// +featureGate=VolumeAttributesClass
+	// +optional
+	VolumeAttributesClassName *string `json:"volumeAttributesClassName,omitempty" protobuf:"bytes,9,opt,name=volumeAttributesClassName"`
 }
```

Comment:
@sunnylovestiramisu 

Does the PV object have this field?

Reply:
Yes, PV should have this field too.

Reply:
Does the status field of the PV need be changed? There's no documentation about this.

Reply:
@sunnylovestiramisu @xing-yang 

In KEP, I cannot find more design details about the `PersistentVolumeSpec.VolumeAttributesClassName` field. 

type PersistentVolumeSpec struct {
  ...
  // Name of VolumeAttributesClass to which this persistent volume belongs. Empty value
  // is not allowed. Use nil to indicate that this volume does not belong to any 
  // VolumeAttributesClass.
  // +optional
  VolumeAttributesClassName *string
  ...
}
```

before I changes the PV definition, I think it's better to discuss the design details first.

1. What's the relationship on the VolumeAttributesClassName field between `PV` and `PVC`?
2. Is support static provisioning for `VolumeAttributesClass`, like `StorageClass`? There's sufficient difference between `VolumeAttributesClass` and `StorageClass`? the StorageClassName is immutable, but the VolumeAttributesClassName is mutable. What's happen to `PV` if the `VolumeAttributesClassName` of a PVC were changed?
3. If the `VolumeAttributesClassName` between `PV` and `PVC` are different but they have the same `StorageClassName`, Can the `PV` be bound to the `PVC`?
4. Can The `VolumeAttributesClassName` field of a PV be reset to nil from a non-nil value?

Reply:
1. For Dynamic Volume Provisioning,  the VolumeAttributesClassName in PV and PVC should be the same
2. We should support static provisioning as well, and plumb in the properties in VAC to the PV at volume creation
3. No, they should not
4. This should not be allowed

There may be some corner cases we should flash out in the KEP, let me update the KEP with more details